src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -369,19 +367,9 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseRep
         // Filtering by courseId
         Course course = courseRepository.findByIdElseThrow(courseId);
         User user = userRepository.getUserWithGroupsAndAuthorities();
-        boolean isAtLeastTutor = authCheckService.isAtLeastTeachingAssistantInCourse(course, user);
-        boolean isAtLeastInstructor = authCheckService.isAtLeastInstructorInCourse(course, user);
-
-        if (!isAtLeastTutor) {
-            throw new AccessForbiddenException("Insufficient permission for these complaints");
-        }
-        if (!isAtLeastInstructor) {
-            // At the moment the complete list of all exam-complaints should only be visible for instructors
-            throw new AccessForbiddenException("Insufficient permission for these complaints");
-        }
-
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
```

Comment:
I think you miss one important aspect of the logic here. Now TAs cannot use this endpoint any more